config/serverless.security.yml

```diff
@@ -9,6 +9,45 @@ xpack.observability.enabled: false
 xpack.observabilityAIAssistant.enabled: false
 xpack.search.notebooks.enabled: false
 
+## Fine-tune the security solution feature privileges.
+xpack.features.overrides:
+  ### Dashboard feature is disabled since it's automatically granted by SIEM feature.
+  dashboard.disabled: true
+  ### Discover feature is disabled since it's automatically granted by SIEM feature.
+  discover.disabled: true
+  ### Machine Learning feature is moved from Analytics category to the Security one as the last item.
+  ml:
+    category: "security"
+    order: 1101
+  ### Security's feature privileges are fine-tuned to grant access to Discover, Dashboard, Maps, and Visualize apps.
+  siem:
+    privileges:
+      ### Security's `All` feature privilege should implicitly grant `All` access to Discover, Dashboard, Maps, and
+      ### Visualize features.
+      all.composedOf:
+        - feature: "discover"
+          privileges: [ "all" ]
+        - feature: "dashboard"
+          privileges: [ "all" ]
+        - feature: "visualize"
+          privileges: [ "all" ]
+        - feature: "maps"
+          privileges: [ "all" ]
```

Comment:
__note:__ I list `visualize` and `maps` directly even though it's already included in `dashboard` to not deal with recursive `composedOf` (it's possible, but I didn't want to add complexity with "topological" sort just yet).

Reply:
Okay, yes recursive `composedOf` sounds tricky, no problem we can keep overrides explicit initially. Thanks for the heads up.